From 4ffdbb752a8ac8cf0538f8162fc1234633c4a5c3 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Fri, 9 Feb 2024 18:14:00 +0100
Subject: [PATCH 508/841] feat(stm32mp21): update STM32 header size for
 STM32MP21

On STM32MP21, the STM32 header v2.3 is used, and the length is 576
bytes, versus 512 bytes for all other STM32MP2 variants.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git db11b37031c7163998b46a699f29ab5afd34e750 feat(stm32mp21): update STM32 header size for STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I8a5778ac8f6331ec083f122150231ede245be514
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/359055
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index be6d683d2..e724f9aec 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -154,8 +154,13 @@ enum ddr_type {
 
 /* Section used inside TF binaries */
 #define STM32MP_PARAM_LOAD_SIZE			U(0x00002400) /* 9 KB for param */
-/* 512 Octets reserved for header */
+#if STM32MP21
+/* 576 Bytes reserved for header */
+#define STM32MP_HEADER_SIZE			U(0x00000240)
+#else /* STM32MP21 */
+/* 512 Bytes reserved for header */
 #define STM32MP_HEADER_SIZE			U(0x00000200)
+#endif /* STM32MP21 */
 #define STM32MP_HEADER_BASE			(STM32MP_SEC_SYSRAM_BASE +	\
 						 STM32MP_PARAM_LOAD_SIZE)
 
-- 
2.47.2

